Solutions/GitLab/Analytic Rules/GitLab_BruteForce.yaml

```diff
@@ -0,0 +1,51 @@
+id: 2238d13a-cf05-4973-a83f-d12a25dbb153
+name: GitLab - Brute-force Attempts
+description: |
+  'This query relies on GitLab Application Logs to get failed logins to highlight brute-force attempts from different IP addresses in a short space of time.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes: 
+      - Syslog
+queryFrequency: 1h
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  let LearningPeriod = 7d; 
+  let BinTime = 1h; 
+  let RunTime = 1h; 
+  let StartTime = 1h; 
+  let NumberOfStds = 3; 
+  let MinThreshold = 10.0; 
+  let EndRunTime = StartTime - RunTime; 
+  let EndLearningTime = StartTime + LearningPeriod;
+  let GitLabFailedLogins = (GitLabApp
+  | where Message contains "Failed Login"
+  | parse kind=regex Message with EventTime ": Failed Login: username=" Username "ip=" IpAddress 
+  | project TimeGenerated, EventTime, Username, IpAddress);
+  GitLabFailedLogins 
+  | where todatetime(EventTime) between (ago(EndLearningTime) .. ago(StartTime)) 
+  | summarize FailedLoginsCountInBinTime = count() by User = Username, bin(todatetime(EventTime), BinTime) 
```

Comment:
Why rename this here, why not just name the field User as part of the parse above in line 29?